### PR TITLE
Adding validation on managementApiEndpoints to be consistent

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
The compromise discussed exposes the managementApiEndpoint for the cloudstack provider in two ways: environment variable, and cluster spec. This decision was made to be future-proof. This PR adds validation on cluster creation to ensure that the two values are the same.

*Testing (if applicable):*
Unit tests pass

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/cloudstack/validator.go [hunk 1]
```diff
@@ -61,6 +62,14 @@ func (v *Validator) ValidateCloudStackDatacenterConfig(ctx context.Context, data
 	if err != nil {
 		return fmt.Errorf("error while checking management api endpoint: %v", err)
 	}
+	execConfig, err := decoder.ParseCloudStackSecret()
+	if err != nil {
+		return fmt.Errorf("parsing cloudstack secret: %v", err)
+	}
+	if execConfig.ManagementUrl != datacenterConfig.Spec.ManagementApiEndpoint {
+		return fmt.Errorf("cloudstack secret management url (%s) differs from cluster spec management url (%s)",
```
Comment (vignesh-goutham) [inline pkg/providers/cloudstack/validator.go:70]: This is probably a great error message. Im wondering if it makes sense to also have a validation if the ManagementApiEndpoint provided on cluster spec to be non-empty, and throw an appropriate message at that point?

Comment (maxdrib) [reply pkg/providers/cloudstack/validator.go:70]: Sure, I can add another check on the provided spec's managementApiEndpoint to make sure it's nonempty

Comment (maxdrib) [reply pkg/providers/cloudstack/validator.go:70]: actually the managementApiEndpoint is a required field per https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go#L36. Does that error on empty string inputs?

**Review by vignesh-goutham — APPROVED** [review]
/lgtm

**Pull request merged**